corehq/apps/reports/templatetags/xform_tags.py

```diff
@@ -168,6 +169,10 @@ def render_form(form, domain, options):
         request and user and request.domain
         and (user.can_edit_data() or user.is_commcare_user())
     )
+    show_edit_options = (
+        user_can_edit
+        and can_edit_form_location(domain, user, form._id)
```

Comment:
this is particularly bad, since we already have the form, then we grab the ID and go get it back in the called function.

another way around this is to have the `can_edit_form_location` take in a `form` or a `form_id` and then you only have to get the doc if you know you don't already have it.